### PR TITLE
[CAMARA] Camara ginkgo stage add the re-capcha

### PR DESCRIPTION
**Pull request by AndreyLykhoman**
**Description:** Added  Goggle's re-captcha to the registration form. I added the re-captcha to two registration form:  One form that is used when FEATURES["ENABLE_COMBINED_LOGIN_REGISTRATION"] is true. And the second one when  FEATURES["ENABLE_COMBINED_LOGIN_REGISTRATION"] is false.

I added the new file: decorators.py. This file contains decorator to validate Goggle's re-captcha.

**Youtrack:** https://youtrack.raccoongang.com/issue/CAMARA-94

**Configuration instructions:** 
In google account create  SECRET_KEY and DATA_SITE_KEY. link: https://developers.google.com/recaptcha/

Add new settings to lms.auth.json with values from google's account:
GOOGLE_RECAPTCHA_SECRET_KEY = #SECRET_KEY
GOOGLE_RECAPTCHA_DATA_SITE_KEY =  #DATA_SITE_KEY

Also, you can set to lms.env.json `"USE_GOOGLE_RECAPTCHA": true`. It means that re-captcha will be in each microsite's registration form.

If you want to use re-captcha only one microsite you should set `"USE_GOOGLE_RECAPTCHA": true` in site configuration in the admin page.

### CODE REVIEW

File: common/djangoapps/student/views.py [hunk 1]
```diff
@@ -122,6 +122,7 @@
     unique_id_for_user
 )
 from student.tasks import send_activation_email
+from student.decorators import check_recaptcha
```
Comment (idegtiarov) [inline common/djangoapps/student/views.py]: Incorrect order of the imported module.

File: common/djangoapps/student/decorators.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            }
+            data = urllib.urlencode(values)
+            req = urllib2.Request(url, data)
+            response = urllib2.urlopen(req)
+            result = json.load(response)
+            # End re-captcha validation
+            if result.get('success'):
+                request.recaptcha_is_valid = True
+            else:
+                request.recaptcha_is_valid = False
```
Comment (idegtiarov) [inline common/djangoapps/student/decorators.py]: This statement seems to be odd and is equal to the recaptcha default value.

File: lms/envs/aws.py [hunk 3]
```diff
@@ -1076,3 +1076,8 @@
 
 # number of seconds from the expiry we consider the token expired
 TOKEN_THRESHOLD = ENV_TOKENS.get('TOKEN_THRESHOLD', 600)
+
+
```
Comment (idegtiarov) [inline lms/envs/aws.py]: Please add comments for the following parameters.

File: lms/templates/student_account/register.underscore [hunk 4]
```diff
@@ -48,6 +48,10 @@
     <% } %>
 
     <%= context.fields %>
+    <% if(context.googleRecaptchaSiteKey) { %>
```
Comment (idegtiarov) [inline lms/templates/student_account/register.underscore]: I would add space after `if`  and before bracket `(`.

File: lms/templates/student_account/register.underscore [hunk 5]
```diff
@@ -48,6 +48,10 @@
...
+        <div class="g-recaptcha" data-sitekey='<%- context.googleRecaptchaSiteKey %>'></div>
```
Comment (idegtiarov) [inline lms/templates/student_account/register.underscore]: I would use double quotes for the `data-sitekey`.

File: openedx/core/djangoapps/user_api/views.py [hunk 6]
```diff
@@ -42,7 +42,7 @@
 from .models import UserPreference, UserProfile
 from .preferences.api import get_country_time_zones, update_email_opt_in
 from .serializers import CountryTimeZoneSerializer, UserPreferenceSerializer, UserSerializer
-
+from student.decorators import check_recaptcha
```
Comment (idegtiarov) [inline openedx/core/djangoapps/user_api/views.py]: Incorrect order of the imported module.

File: common/djangoapps/student/decorators.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import urllib2
+import json
+from functools import wraps
+from django.conf import settings
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+
+def check_recaptcha(view_func):
+    """
+    Check re-captcha.
```
[the remark this answers is not in the document]
Comment (AndreyLykhoman) [reply common/djangoapps/student/decorators.py]: Done

File: lms/templates/student_account/register.underscore [hunk 8]
```diff
@@ -48,8 +48,8 @@
     <% } %>
 
     <%= context.fields %>
-    <% if(context.googleRecaptchaSiteKey) { %>
-        <div class="g-recaptcha" data-sitekey='<%- context.googleRecaptchaSiteKey %>'></div>
+    <% if (context.googleRecaptchaSiteKey) { %>
+        <div class="g-recaptcha" data-sitekey="<%- context.googleRecaptchaSiteKey %>""></div>
```
Comment (idegtiarov) [inline lms/templates/student_account/register.underscore]: It seems one double quote at the end of the line is odd.

**Pull request merged**